crates/jstz_kernel/src/riscv_wpt_test_kernel.rs

```diff
@@ -0,0 +1,96 @@
+#![cfg(feature = "riscv_wpt_test_kernel")]
+
+use tezos_smart_rollup::{host::Runtime, prelude::debug_msg};
+
+use crate::inbox::*;
+use jstz_core::kv::Transaction;
+use jstz_crypto::{hash::Hash, smart_function_hash::SmartFunctionHash};
+use jstz_proto::operation::internal::InboxId;
+use jstz_proto::operation::{Content, Operation};
+use jstz_runtime::wpt::{init_runtime, TestHarnessReport};
+use tezos_crypto_rs::hash::ContractKt1Hash;
+use tezos_crypto_rs::hash::SmartRollupHash;
+use tezos_smart_rollup_mock::MockHost;
+
+fn read_message(
+    rt: &mut impl Runtime,
+    ticketer: &ContractKt1Hash,
+) -> Option<ParsedInboxMessage> {
+    let input = match rt.read_input() {
+        Ok(input) => match input {
+            Some(input) => input,
+            None => {
+                return None;
+            }
+        },
+        Err(_e) => {
+            return None;
+        }
+    };
+    let jstz_rollup_address =
+        SmartRollupHash::from_base58_check("sr1BxufbqiHt3dn6ahV6eZk9xBD6XV1fYowr")
+            .unwrap();
+    let inbox_id = InboxId {
+        l1_level: input.level,
+        l1_message_id: input.id,
+    };
+    parse_inbox_message(rt, inbox_id, input.as_ref(), ticketer, &jstz_rollup_address)
+}
+
+pub fn entry(rt: &mut impl Runtime) {
+    let mut tx = Transaction::default();
+    tx.begin();
+    let mut host = MockHost::default();
+    host.set_debug_handler(std::io::empty());
+    let ticketer =
+        SmartFunctionHash::from_base58("KT1RJ6PbjHpwc3M5rw5s2Nbmefwbuwbdxton").unwrap();
+
+    let mut source = String::new();
+
+    loop {
+        match read_message(rt, &ticketer) {
+            Some(message) => match message {
+                ParsedInboxMessage::JstzMessage(Message::External(signed_operation)) => {
+                    let operation: Operation = signed_operation.into();
+                    match operation.content {
+                        Content::DeployFunction(deploy_function) => {
+                            if deploy_function.function_code.to_string() == "STOP" {
+                                break;
+                            }
+                            source += &deploy_function.function_code.to_string();
+                        }
+                        _ => {}
+                    }
+                }
+                _ => {}
+            },
+            None => {
+                break;
+            }
+        }
+    }
```

Comment:
Single match paths should almost always be `if lets`

```suggestion
   while let Some(ParsedInboxMessage::JstzMessage(Message::External(signed_operation))) =
        read_message(rt, &ticketer)
    {
        let operation: Operation = signed_operation.into();
        if let Content::DeployFunction(deploy_function) = operation.content {
            if deploy_function.function_code.to_string() == "STOP" {
                break;
            }
            source += &deploy_function.function_code.to_string();
        }
    }
```

Reply:
not exactly like this since you can also have an internal message, but I've updated it to the extend its possible